src/test/ui/consts/const-int-checked.rs

```diff
@@ -0,0 +1,57 @@
+// run-pass
+#![feature(const_int_checked)]
+
+macro_rules! assert_same_const {
+    ($(const $ident:ident: $ty:ty = $exp:expr;)+) => {
+        $(const $ident: $ty = $exp;)+
+
+        pub fn main() {
+            $(assert_eq!($exp, $ident);)+
```

Comment:
In order to prevent promotion from making this test useless because it ends up promoting the assert argument 
```suggestion
            $(let tmp = $exp; assert_eq!(tmp, $ident);)+
```